wcfsetup/install/files/lib/system/file/upload/UploadFile.class.php

```diff
@@ -219,9 +219,11 @@ public function setProcessed($newLocation)
      *
      * @param string $link
      */
-    public function setImageLink($link)
+    public function setImageLink(string $link): UploadFile
```

Comment:
Adding a new return value/return type is a breaking change, as this class is not `final`. Additionally this is inconsistent with e.g. `setProcessed()` not returning anything at all.

Reply:
The return type definition doesn't matter for me; I was used to do so since we're doing it strictly everywhere for the (internal code of the) new shop. ^^
Important for me is that it returns itself for a usage like that:
```php
	public function getFileUploadFileLocations() : array {
		$path = $this->getLocation();
		
		if (\file_exists($path)) {
			return [
				(new UploadFile($path, $this->getFilename()))
					->setImageLink($this->getLink()),
			];
		}
		
		return [$path];
	}
```

However the phpDoc is missing the return type; totally forgot that.

Reply:
>  Important for me is that it returns itself for a usage like that:

This would be pretty inconsistent across the setters in the `UploadFile` class, because not all of them `return $this;`

> Important for me is that it returns itself for a usage like that:

In any case I don't really see the benefit of such an interface here: It's not like PHP DDL/form builder where you need to define _large amounts_ of “static data” (i.e. a hardcoded list of 10+ fields).

You are actually creating a dynamic object here – just use a temporary variable.